### PR TITLE
feat(shipper): Add Auspost

### LINKED ISSUE #548: [Feature Request] New Shipper (Australia Post)
**Describe the solution you'd like**

I'm afraid I'm not very skilled with Python so I've tried to provide as much detail as possible.
Australia Post doesn't have images included in the email but has several different templates they use.

Screenshots of the primary templates are below.

The email address that is used for these emails is: `noreply@notifications.auspost.com.au`

**Additional context**
Subject: Your delivery from `xxx` is being prepared
Preparing

Subject: Your delivery is on its way
On its way

Subject: Your delivery is coming today
Coming Today

Subject: Your shipment has been delivered
Has been delivered

### PR DESCRIPTION
**Pull request by sebasptsch**
fixes #548

### CODE REVIEW

**Comment by firstof9** [comment]
Mind doing me a favor and fix the test?
https://github.com/moralmunky/Home-Assistant-Mail-And-Packages/blob/479ab3cdde9c78c8996863f88fa1abe3f7d7f00c/tests/test_config_flow.py#L865-L868

change to:
```python
    assert "Error connecting into IMAP Server:" in caplog.text
```

**Comment by sebasptsch** [comment]
Hopefully I've been able to rewrite the tests correctly for the `auspost` shipper. Just wondering if I could get some help figuring out the tracking info. The test emails have tracking numbers in them but there seems to be no reliable pattern between them.

**Comment by firstof9** [comment]
> Just wondering if I could get some help figuring out the tracking info.

Sure thing do you have the plain text version of the emails or do they only send in HTML format?

File: custom_components/mail_and_packages/const.py [hunk 1]
```diff
@@ -282,6 +291,16 @@
         "package(s)",
         "mdi:package-variant-closed",
     ],
+    "auspost_delivered": [
+        "Mail Packages Delivered",
+        "package(s)",
+        "mdi:package-variant",
+    ],
+    "auspost_transit": [
```
Comment (firstof9) [inline custom_components/mail_and_packages/const.py:299]:
```suggestion
    "auspost_delivering": [
```

File: custom_components/mail_and_packages/const.py [hunk 2]
```diff
@@ -188,6 +188,15 @@
...
     "royal_tracking": {"pattern": ["[A-Za-z]{2}[0-9]{9}GB"]},
+    "auspost_delivered": {
+        "email": ["noreply@notifications.auspost.com.au"],
+        "subject": ["Your shipment has been delivered"],
+    },
+    "auspost_delivering": {
+        "email": ["noreply@notifications.auspost.com.au"],
+        "subject": ["Your delivery is coming today"],
+    },
+    "auspost_packages": {},
```
Comment (firstof9) [inline custom_components/mail_and_packages/const.py:199]:
```suggestion
    "auspost_packages": {},
    "auspost_tracking": {},
```

**Review by firstof9 — CHANGES_REQUESTED** [review]

**Comment by sebasptsch** [comment]
> > Just wondering if I could get some help figuring out the tracking info.
> 
> Sure thing do you have the plain text version of the emails or do they only send in HTML format?

I added some emails to the test emails folder which might be usable.

File: custom_components/mail_and_packages/const.py (hunk 2 above)
Comment (firstof9) [reply custom_components/mail_and_packages/const.py:199]: Based on the provided emails it looks like this would be the pattern to search for.

Comment (sebasptsch) [reply custom_components/mail_and_packages/const.py:199]: I've added a few more combinations (I'm not sure if this is how the OR operator is used in this context)

File: tests/const.py [hunk 3]
```diff
@@ -30,6 +30,9 @@
         "dhl_delivering",
         "dhl_packages",
         "amazon_delivered",
+        "auspost_delivered",
+        "auspost_packages",
+        "auspost_delivering"
```
Comment (firstof9) [inline tests/const.py:35]:
```suggestion
        "auspost_delivering",
```

File: tests/const.py [hunk 4]
```diff
@@ -76,6 +79,9 @@
         "usps_delivering",
         "usps_mail",
         "usps_packages",
+        "auspost_delivered",
+        "auspost_delivering"
+        "auspost_packages"
```
Comment (firstof9) [inline tests/const.py:84]:
```suggestion
        "auspost_packages",
```

File: tests/const.py [hunk 5]
```diff
@@ -122,6 +128,9 @@
         "usps_delivering",
         "usps_mail",
         "usps_packages",
+        "auspost_delivered",
+        "auspost_delivering"
```
Comment (firstof9) [inline tests/const.py:132]:
```suggestion
        "auspost_delivering",
```

File: tests/const.py [hunk 6]
```diff
@@ -122,6 +128,9 @@
...
+        "auspost_packages"
```
Comment (firstof9) [inline tests/const.py:133]:
```suggestion
        "auspost_packages",
```

File: tests/const.py [hunk 7]
```diff
@@ -168,6 +177,9 @@
         "usps_packages",
         "zpackages_delivered",
         "zpackages_transit",
+        "auspost_delivered",
+        "auspost_delivering"
```
Comment (firstof9) [inline tests/const.py:181]:
```suggestion
        "auspost_delivering",
```

File: tests/const.py [hunk 8]
```diff
@@ -215,6 +227,9 @@
...
```
Comment (firstof9) [inline tests/const.py:231]:
```suggestion
        "auspost_delivering",
```

File: tests/const.py [hunk 9]
```diff
@@ -258,6 +273,9 @@
...
```
Comment (firstof9) [inline tests/const.py:277]:
```suggestion
        "auspost_delivering",
```

File: tests/const.py [hunk 10]
```diff
@@ -299,6 +317,9 @@
...
```
Comment (firstof9) [inline tests/const.py:321]:
```suggestion
        "auspost_delivering",
```

File: tests/const.py [hunk 11]
```diff
@@ -339,6 +360,9 @@
...
```
Comment (firstof9) [inline tests/const.py:364]:
```suggestion
        "auspost_delivering",
```

File: tests/const.py [hunk 12]
```diff
@@ -415,6 +442,9 @@
         "usps_delivering",
         "usps_mail",
         "usps_packages",
+        "auspost_delivered",
+        "auspost_delivering"
```
Comment (firstof9) [inline tests/const.py:446]:
```suggestion
        "auspost_delivering",
```

File: tests/const.py [hunk 13]
```diff
@@ -461,6 +491,9 @@
...
```
Comment (firstof9) [inline tests/const.py:495]:
```suggestion
        "auspost_delivering",
```

File: tests/const.py [hunk 14]
```diff
@@ -508,6 +541,9 @@
         "usps_packages",
         "zpackages_delivered",
         "zpackages_transit",
+        "auspost_delivered",
+        "auspost_delivering"
```
Comment (firstof9) [inline tests/const.py:545]:
```suggestion
        "auspost_delivering",
```

File: tests/test_config_flow.py [hunk 15]
```diff
@@ -56,7 +56,10 @@
                     "dhl_delivered",
                     "dhl_delivering",
                     "dhl_packages",
-                    "amazon_delivered",
+"amazon_delivered",                   
+                    "auspost_delivered",
+                    "auspost_delivering"
```
Comment (firstof9) [inline tests/test_config_flow.py:61]:
```suggestion
                    "auspost_delivering",
```

File: tests/test_config_flow.py [hunk 16]
```diff
@@ -56,7 +56,10 @@
                     "dhl_delivered",
                     "dhl_delivering",
                     "dhl_packages",
-                    "amazon_delivered",
+"amazon_delivered",
```
Comment (firstof9) [inline tests/test_config_flow.py:59]:
```suggestion
                    "amazon_delivered",
```

File: tests/test_config_flow.py [hunk 17]
```diff
@@ -96,7 +99,10 @@
...
```
Comment (firstof9) [inline tests/test_config_flow.py:102]:
```suggestion
                    "amazon_delivered",
```

File: tests/test_config_flow.py [hunk 18]
```diff
@@ -96,7 +99,10 @@
                     "dhl_delivered",
                     "dhl_delivering",
                     "dhl_packages",
-                    "amazon_delivered",
+"amazon_delivered",                   
+                    "auspost_delivered",
+                    "auspost_delivering"
```
Comment (firstof9) [inline tests/test_config_flow.py:104]:
```suggestion
                    "auspost_delivering",
```

File: tests/test_config_flow.py [hunk 19]
```diff
@@ -200,7 +206,10 @@ async def test_form(
                     "dhl_delivered",
                     "dhl_delivering",
                     "dhl_packages",
-                    "amazon_delivered",
+"amazon_delivered",
```
Comment (firstof9) [inline tests/test_config_flow.py:209]:
```suggestion
                    "amazon_delivered",
```

File: tests/test_config_flow.py [hunk 20]
```diff
@@ -200,7 +206,10 @@ async def test_form(
                     "dhl_delivered",
                     "dhl_delivering",
                     "dhl_packages",
-                    "amazon_delivered",
+"amazon_delivered",                   
+                    "auspost_delivered",
+                    "auspost_delivering"
```
Comment (firstof9) [inline tests/test_config_flow.py:211]:
```suggestion
                    "auspost_delivering",
```

File: tests/test_config_flow.py [hunk 21]
```diff
@@ -240,7 +249,10 @@ async def test_form(
                     "dhl_delivered",
                     "dhl_delivering",
                     "dhl_packages",
-                    "amazon_delivered",
+"amazon_delivered",
```
Comment (firstof9) [inline tests/test_config_flow.py:252]:
```suggestion
                    "amazon_delivered",
```

File: tests/test_config_flow.py [hunk 22]
```diff
@@ -240,7 +249,10 @@ async def test_form(
                     "dhl_delivered",
                     "dhl_delivering",
                     "dhl_packages",
-                    "amazon_delivered",
+"amazon_delivered",                   
+                    "auspost_delivered",
+                    "auspost_delivering"
```
Comment (firstof9) [inline tests/test_config_flow.py:254]:
```suggestion
                    "auspost_delivering",
```

File: tests/test_config_flow.py [hunk 23]
```diff
@@ -381,7 +393,10 @@ async def test_form_connection_error(input_1, step_id_2, hass, mock_imap):
                     "dhl_delivered",
                     "dhl_delivering",
                     "dhl_packages",
-                    "amazon_delivered",
+                    "amazon_delivered",                   
+                    "auspost_delivered",
+                    "auspost_delivering"
```
Comment (firstof9) [inline tests/test_config_flow.py:398]:
```suggestion
                    "auspost_delivering",
```

File: tests/test_config_flow.py [hunk 24]
```diff
@@ -381,7 +393,10 @@ async def test_form_connection_error(input_1, step_id_2, hass, mock_imap):
...
+                    "auspost_packages"
```
Comment (firstof9) [inline tests/test_config_flow.py:399]:
```suggestion
                    "auspost_packages",
```

File: tests/test_config_flow.py [hunk 25]
```diff
@@ -414,7 +429,10 @@ async def test_form_connection_error(input_1, step_id_2, hass, mock_imap):
                     "dhl_delivered",
                     "dhl_delivering",
                     "dhl_packages",
-                    "amazon_delivered",
+                    "amazon_delivered",                   
+                    "auspost_delivered",
+                    "auspost_delivering"
```
Comment (firstof9) [inline tests/test_config_flow.py:434]:
```suggestion
                    "auspost_delivering",
```

File: tests/test_config_flow.py [hunk 26]
```diff
@@ -414,7 +429,10 @@ async def test_form_connection_error(input_1, step_id_2, hass, mock_imap):
...
+                    "auspost_packages"
```
Comment (firstof9) [inline tests/test_config_flow.py:435]:
```suggestion
                    "auspost_packages",
```

File: tests/test_config_flow.py [hunk 27]
```diff
@@ -498,7 +516,10 @@ async def test_form_invalid_ffmpeg(
                     "dhl_delivered",
                     "dhl_delivering",
                     "dhl_packages",
-                    "amazon_delivered",
+                    "amazon_delivered",                   
+                    "auspost_delivered",
+                    "auspost_delivering"
```
Comment (firstof9) [inline tests/test_config_flow.py:521]:
```suggestion
                    "auspost_delivering",
```

File: tests/test_config_flow.py [hunk 28]
```diff
@@ -626,7 +650,10 @@ async def test_form_index_error(
...
```
Comment (firstof9) [inline tests/test_config_flow.py:655]:
```suggestion
                    "auspost_delivering",
```

File: tests/test_config_flow.py [hunk 29]
```diff
@@ -661,7 +688,10 @@ async def test_form_index_error(
...
```
Comment (firstof9) [inline tests/test_config_flow.py:693]:
```suggestion
                    "auspost_delivering",
```

File: tests/test_config_flow.py [hunk 30]
```diff
@@ -753,7 +783,10 @@ async def test_form_index_error_2(
...
```
Comment (firstof9) [inline tests/test_config_flow.py:788]:
```suggestion
                    "auspost_delivering",
```

File: tests/test_config_flow.py [hunk 31]
```diff
@@ -788,7 +821,10 @@ async def test_form_index_error_2(
...
```
Comment (firstof9) [inline tests/test_config_flow.py:826]:
```suggestion
                    "auspost_delivering",
```

File: tests/test_config_flow.py [hunk 32]
```diff
@@ -911,7 +944,10 @@ async def test_imap_login_error(mock_imap_login_error, caplog):
...
```
Comment (firstof9) [inline tests/test_config_flow.py:949]:
```suggestion
                    "auspost_delivering",
```

File: tests/test_config_flow.py [hunk 33]
```diff
@@ -954,7 +990,10 @@ async def test_imap_login_error(mock_imap_login_error, caplog):
...
```
Comment (firstof9) [inline tests/test_config_flow.py:995]:
```suggestion
                    "auspost_delivering",
```

File: tests/test_config_flow.py [hunk 34]
```diff
@@ -1069,7 +1108,10 @@ async def test_options_flow(
...
```
Comment (firstof9) [inline tests/test_config_flow.py:1113]:
```suggestion
                    "auspost_delivering",
```

File: tests/test_config_flow.py [hunk 35]
```diff
@@ -1112,7 +1154,10 @@ async def test_options_flow(
...
```
Comment (firstof9) [inline tests/test_config_flow.py:1159]:
```suggestion
                    "auspost_delivering",
```

File: tests/test_config_flow.py [hunk 36]
```diff
@@ -1280,7 +1325,10 @@ async def test_options_flow_connection_error(
...
```
Comment (firstof9) [inline tests/test_config_flow.py:1330]:
```suggestion
                    "auspost_delivering",
```

File: tests/test_config_flow.py [hunk 37]
```diff
@@ -1315,7 +1363,10 @@ async def test_options_flow_connection_error(
...
```
Comment (firstof9) [inline tests/test_config_flow.py:1368]:
```suggestion
                    "auspost_delivering",
```

File: tests/test_config_flow.py [hunk 38]
```diff
@@ -1410,7 +1461,10 @@ async def test_options_flow_invalid_ffmpeg(
...
```
Comment (firstof9) [inline tests/test_config_flow.py:1466]:
```suggestion
                    "auspost_delivering",
```

File: tests/test_config_flow.py [hunk 39]
```diff
@@ -1445,7 +1499,10 @@ async def test_options_flow_invalid_ffmpeg(
...
```
Comment (firstof9) [inline tests/test_config_flow.py:1504]:
```suggestion
                    "auspost_delivering",
```

File: tests/test_config_flow.py [hunk 40]
```diff
@@ -1541,7 +1598,10 @@ async def test_options_flow_index_error(
...
```
Comment (firstof9) [inline tests/test_config_flow.py:1603]:
```suggestion
                    "auspost_delivering",
```

File: tests/test_config_flow.py [hunk 41]
```diff
@@ -1576,7 +1636,10 @@ async def test_options_flow_index_error(
...
```
Comment (firstof9) [inline tests/test_config_flow.py:1641]:
```suggestion
                    "auspost_delivering",
```

File: tests/test_config_flow.py [hunk 42]
```diff
@@ -1672,7 +1735,10 @@ async def test_options_flow_index_error_2(
...
```
Comment (firstof9) [inline tests/test_config_flow.py:1740]:
```suggestion
                    "auspost_delivering",
```

File: tests/test_config_flow.py [hunk 43]
```diff
@@ -1707,7 +1773,10 @@ async def test_options_flow_index_error_2(
...
```
Comment (firstof9) [inline tests/test_config_flow.py:1778]:
```suggestion
                    "auspost_delivering",
```

File: tests/test_config_flow.py [hunk 44]
```diff
@@ -1802,7 +1871,10 @@ async def test_options_flow_mailbox_format2(
...
```
Comment (firstof9) [inline tests/test_config_flow.py:1876]:
```suggestion
                    "auspost_delivering",
```

File: tests/test_config_flow.py [hunk 45]
```diff
@@ -1840,7 +1912,10 @@ async def test_options_flow_mailbox_format2(
...
```
Comment (firstof9) [inline tests/test_config_flow.py:1917]:
```suggestion
                    "auspost_delivering",
```

File: tests/test_config_flow.py [hunk 46]
```diff
@@ -1942,7 +2017,10 @@ async def test_options_flow_bad(
...
```
Comment (firstof9) [inline tests/test_config_flow.py:2022]:
```suggestion
                    "auspost_delivering",
```

File: tests/test_config_flow.py [hunk 47]
```diff
@@ -2027,7 +2105,10 @@ async def test_form_amazon_error(
...
```
Comment (firstof9) [inline tests/test_config_flow.py:2110]:
```suggestion
                    "auspost_delivering",
```

**Review by firstof9 — CHANGES_REQUESTED** [review]
Just a few missing `,`s

**Comment by sebasptsch** [comment]
Sorry about all the minor errors. I don't have a python environment setup so it's not picking up on these things and I can't test locally.

**Comment by firstof9** [comment]
No worries, here's a tip, on github you can hit `.` and it'll open VS Code in your browser to make editing code far easier.

**Comment by sebasptsch** [comment]
> 
> 
> No worries, here's a tip, on github you can hit `.` and it'll open VS Code in your browser to make editing code far easier.

That's what I've been doing but aparently Python isn't fully supported on the web version of VSCode.

File: custom_components/mail_and_packages/const.py [hunk 48]
```diff
@@ -282,6 +292,16 @@
...
+    "auspost_delivered": [
+        "Mail Packages Delivered",
+        "package(s)",
+        "mdi:package-variant",
+    ],
+    "auspost_delivering": [
+        "Mail Packages In Transit",
+        "package(s)",
+        "mdi:truck-delivery",
+    ],
```
Comment (firstof9) [inline custom_components/mail_and_packages/const.py:304]:
```suggestion
    ],
    "auspost_packages": [
        "Mail AUSPost Packages",
        "package(s)",
        "mdi:package-variant-closed",
    ],    
```

File: custom_components/mail_and_packages/const.py [hunk 49]
```diff
@@ -282,6 +292,16 @@
         "package(s)",
         "mdi:package-variant-closed",
     ],
+    "auspost_delivered": [
+        "Mail Packages Delivered",
```
Comment (firstof9) [inline custom_components/mail_and_packages/const.py:296]:
```suggestion
        "Mail AUSPost Delivered",
```

File: custom_components/mail_and_packages/const.py [hunk 50]
```diff
@@ -282,6 +292,16 @@
...
+        "package(s)",
+        "mdi:package-variant",
+    ],
+    "auspost_delivering": [
+        "Mail Packages In Transit",
```
Comment (firstof9) [inline custom_components/mail_and_packages/const.py:301]:
```suggestion
        "Mail AUSPost Packages In Transit",
```

**Review by firstof9 — CHANGES_REQUESTED** [review]
I think these last few changes should fix everything up.

**Comment by firstof9** [comment]
Looks like the tests are all fixed now, just waiting for all the checks.

**Review by firstof9 — APPROVED** [review]

**Comment by firstof9** [comment]
Thanks! 👍

**Pull request merged**